packages/flutter/test/material/app_test.dart

```diff
@@ -812,6 +812,33 @@ void main() {
     tester.binding.window.accessibilityFeaturesTestValue = null;
   });
 
+  testWidgets('MaterialApp uses dark theme when no high contrast dark theme is provided', (WidgetTester tester) async {
+    tester.binding.window.platformBrightnessTestValue = Brightness.dark;
+    tester.binding.window.accessibilityFeaturesTestValue = MockAccessibilityFeature();
+
+    ThemeData appliedTheme;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+          primaryColor: Colors.lightBlue,
+        ),
+        darkTheme: ThemeData(
+          primaryColor: Colors.lightGreen,
+        ),
+        home: Builder(
+          builder: (BuildContext context) {
+            appliedTheme = Theme.of(context);
+            return const SizedBox();
+          },
+        ),
+      ),
+    );
+
+    expect(appliedTheme.primaryColor, Colors.lightGreen);
+    tester.binding.window.accessibilityFeaturesTestValue = null;
```

Comment:
Do you also set `tester.binding.window.platformBrightnessTestValue = Brightness.dark;` to null?

Reply:
Yes I should, done